src/Catel.MVVM/Catel.MVVM.Xamarin.Forms/IoC/ServiceLocatorDependencyRegistrationManager.cs

```diff
@@ -0,0 +1,101 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="ServiceLocatorDependencyRegistrationManager.cs" company="Catel development team">
+//   Copyright (c) 2008 - 2017 Catel development team. All rights reserved.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace Catel.IoC
+{
+    using System;
+    using System.Linq;
+    using System.Reflection;
+    using Catel.Logging;
+    using Catel.Reflection;
+    using global::Xamarin.Forms;
+
+    /// <summary>
+    /// The service locator dependency registration manager class.
+    /// </summary>
+    public class ServiceLocatorDependencyRegistrationManager
+    {
+        /// <summary>
+        /// The log.
+        /// </summary>
+        private static readonly ILog Log = LogManager.GetCurrentClassLogger();
+
+        /// <summary>
+        /// An array of object with a single null element
+        /// </summary>
+        private static readonly object[] ArrayOfObjectWithSingleNullElement = { null };
+
+        /// <summary>
+        /// The dependency attribute Get method info.
+        /// </summary>
+        private static readonly MethodInfo DependencyServiceGetMethodInfo = typeof(DependencyService).GetMethodEx("Get", BindingFlags.Static | BindingFlags.Public);
+
+        /// <summary>
+        /// The dependency attribute Implementor property info.
+        /// </summary>
+        private static readonly PropertyInfo DependencyAttributeImplementorPropertyInfo = typeof(DependencyAttribute).GetPropertyEx("Implementor", BindingFlags.Instance | BindingFlags.NonPublic);
+
+        /// <summary>
+        /// The service locator.
+        /// </summary>
+        private readonly IServiceLocator _serviceLocator;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ServiceLocatorDependencyRegistrationManager" /> class.
+        /// </summary>
+        /// <param name="serviceLocator">The service locator</param>
+        public ServiceLocatorDependencyRegistrationManager(IServiceLocator serviceLocator)
+        {
+            Argument.IsNotNull(()=> serviceLocator);
+
+            _serviceLocator = serviceLocator;
+        }
+
+        /// <summary>
+        /// Initialize
+        /// </summary>
+        public void Initialize()
+        {
+            //// TODO: Use type cache insted?
```

Comment:
Create a GitHub ticket so we won't forget about this one. TypeCache is the recommended approach.